File: Codes/Haskell/2.hs
main = mapM_ putStrLn $ fbz [(2, "Fuzz"), (3, "Fizz"), (5, "Buzz"), (7, "Bizz")] [1 .. 1000]

fbz tl = map (\n -> let sl = [b | (a, b) <- tl, mod n a == 0] in if (sl == []) then show n else concat sl)

File: Codes/C++/2.cpp
#include <iostream>
using namespace std;

void bizzBuzz()
{
	for(int i = 1; i <= 100; i++)
	{
		bool print_number = true;
		if(i % 3 == 0)
		{
			cout << "Bizz" << ' ';
			print_number = false;
		}
		if (i % 5 == 0)
		{
			cout << "Buzz" << ' ';
			print_number = false;
		}
		if (i % 7 == 0)
		{
			cout << "Woof" << ' ';
			print_number = false;
		}
		if (print_number)
			cout << "(" << i << ")" << ' ';

		// cout << endl;
	}
}

int main()
{
	bizzBuzz();

	return 0;
}
